flink-runtime/src/main/java/org/apache/flink/runtime/operators/coordination/OperatorCoordinatorHolder.java

```diff
@@ -397,12 +580,19 @@ public void abortCurrentTriggering() {
         // we can remove the delegation once the checkpoint coordinator runs fully in the
         // scheduler's main thread executor
         mainThreadExecutor.execute(
-                () ->
-                        subtaskGatewayMap
-                                .values()
-                                .forEach(
-                                        SubtaskGatewayImpl
-                                                ::openGatewayAndUnmarkLastCheckpointIfAny));
+                () -> {
+                    abortAcknowledgeCloseGatewayFutures(
```

Comment:
Yes. There is no guarantee from the documents that `abortCurrentTriggering` must be invoked with `notifyCheckpointAborted`. From the source code of `CheckpointCoordinator`, it can be found that if the checkpoint fails during the initialization process, `abortCurrentTriggering` would be invoked while `notifyCheckpointAborted` would not.